### ISSUE TITLE
FirefoxDriver does not close FirefoxPortable when running close or quit method

### ISSUE THREAD

**Issue by tkleiber** [SeleniumHQ/selenium#618]
The test is successful, but FirefoxPortable remains open after test.
Reproduced with Windows7 Enterprise, FirefoxPortable_38.0.5_German, selenium 2.45.0 and JUnit 4.11).

Please try following to reproduce:

import java.io.File;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

import org.openqa.selenium.firefox.FirefoxBinary;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxProfile;
import org.openqa.selenium.remote.RemoteWebDriver;

public class UnitTest {
    @Test
    public void testCloseAndQuitFirefoxPortable() throws Exception {
        FirefoxProfile ffProfile = new FirefoxProfile();
        // The binary is installed from latest german version FirefoxPortable_38.0.5_German.paf.exe
        FirefoxBinary ffBinary = new FirefoxBinary(new File("X:/Applications/FirefoxPortable/FirefoxPortable.exe"));
        RemoteWebDriver driver = new FirefoxDriver(ffBinary, ffProfile);
        driver.get("http://www.google.de");
        assertEquals("Google", driver.getTitle());
        // Close the current window, quitting the browser if it's the last window currently open.
        // TODO but is does not
        driver.close();
        // Quits this driver, closing every associated window
        // TODO but is does not
        driver.quit();
    }
}

**Comment by lukeis** [comment]
firefox portable isn't in our normal testing suite (and I don't think it ever will be).

When you run this, is there any output in Firefox's developer console?

**Comment by tkleiber** [comment]
There is only output, which comes when normal starting www.google.de too, nothing special to the close/quit event:

This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] www.google.de
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] start-of-the-2015-fifa-womens-world-cup-5164271868575744-res.png
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] start-of-the-2015-fifa-womens-world-cup-5164271868575744-hp.jpg
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] logo-myaccount-hpp-48px.png
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] i2_9ef0f6fa.png
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] rs=ACT90oHkZLjkTpe-dKujRgyS29shXkF_4Q
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] rs=AItRSTO2fhyu0-p_Vkan7L4YwNDwQhfB_w
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] rs=ACT90oHkZLjkTpe-dKujRgyS29shXkF_4Q
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] cb=gapi.loaded_0
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] gen_204
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] 47e2987e09c332a4.js
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] tia.png
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] nav_logo195.png
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] favicon.ico
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] gen_204
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] 6.gif
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] 6.gif

**Comment by barancev** [comment]
What flavor of Firefox portable do you use? Where can I download it to reproduce the issue? Is it reproduced on the German version only?

**Comment by lukeis** [comment]
so, firefox portable is essentially a hack of mozilla firefox to run specifically on a usb stick on windows?

It seems hard to justify using this version of firefox over the regular installed version when running automation. I'm inclined to close this issue as Won't Fix  (no, selenium doesn't pretend to actually support all browsers nor all versions of browsers...).

I wouldn't reject a patch to fix this, but we're not going to ever be running tests against it and will have no way to guarantee future versions of the browser will continue to work.

**Comment by mfowlewebs** [comment]
Our team is seeing something very similar in our QA environments, using normal Firefox 38.0.5 (not FirefoxPortable). I thought I'd at least try running something like the provided example, and see what happened, see if this easily reproduced. I added a pom and got rid of junit, and here's the provided test case- https://github.com/mfowlewebs/selenium-noquit 

Alas, no replication. We've switched from using Selenium 2.x to Selenium 2.45 for now, we're having what sounds like these same symptoms without FirefoxPortable (with regular Firefox), but I don't think we have any bead on how to get a reproducible test case up to begin to get attention here. v0v
